*** Create_TRD_Geometry_v14a300.C	2013-12-06 17:04:35.084573025 +0100
--- Create_TRD_Geometry_v14a100.C	2013-12-06 17:07:04.105153920 +0100
***************
*** 72,78 ****
  #include <iostream>
  
  // Name of output file with geometry
! const TString geoVersion   = "trd_v14a300";  // "trd_v14a100";
  const TString FileNameSim  = geoVersion + ".geo.root";
  const TString FileNameGeo  = geoVersion + "_geo.root";
  const TString FileNameInfo = geoVersion + ".geo.info";
--- 72,78 ----
  #include <iostream>
  
  // Name of output file with geometry
! const TString geoVersion   = "trd_v14a100";  // "trd_v14a100";
  const TString FileNameSim  = geoVersion + ".geo.root";
  const TString FileNameGeo  = geoVersion + "_geo.root";
  const TString FileNameInfo = geoVersion + ".geo.info";
***************
*** 128,136 ****
  //const Int_t    ShowLayer[MaxLayers] = { 1, 1, 0, 0, 0, 0, 0, 0, 0, 0 };  // SIS100-2l  // 1: plot, 0: hide
  //const Int_t    ShowLayer[MaxLayers] = { 1, 1, 1, 0, 0, 0, 0, 0, 0, 0 };  // SIS100-3l  // 1: plot, 0: hide
  //
! //const Int_t    ShowLayer[MaxLayers] = { 1, 1, 1, 1, 0, 0, 0, 0, 0, 0 };  // SIS100-4l  // 1: plot, 0: hide
  //const Int_t    ShowLayer[MaxLayers] = { 0, 0, 0, 0, 1, 1, 1, 1, 1, 1 };  // SIS300-mu  // 1: plot, 0: hide
! const Int_t    ShowLayer[MaxLayers] = { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 };  // SIS300-e   // 1: plot, 0: hide
  
  Int_t    PlaneId[MaxLayers]; // automatically filled with layer ID
  
--- 128,136 ----
  //const Int_t    ShowLayer[MaxLayers] = { 1, 1, 0, 0, 0, 0, 0, 0, 0, 0 };  // SIS100-2l  // 1: plot, 0: hide
  //const Int_t    ShowLayer[MaxLayers] = { 1, 1, 1, 0, 0, 0, 0, 0, 0, 0 };  // SIS100-3l  // 1: plot, 0: hide
  //
! const Int_t    ShowLayer[MaxLayers] = { 1, 1, 1, 1, 0, 0, 0, 0, 0, 0 };  // SIS100-4l  // 1: plot, 0: hide
  //const Int_t    ShowLayer[MaxLayers] = { 0, 0, 0, 0, 1, 1, 1, 1, 1, 1 };  // SIS300-mu  // 1: plot, 0: hide
! //const Int_t    ShowLayer[MaxLayers] = { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 };  // SIS300-e   // 1: plot, 0: hide
  
  Int_t    PlaneId[MaxLayers]; // automatically filled with layer ID
  
***************
*** 371,377 ****
  void dump_info_file();
  
  
! void Create_TRD_Geometry_v14a300() {
    // Load the necessary FairRoot libraries 
    gROOT->LoadMacro("$VMCWORKDIR/gconfig/basiclibs.C");
    basiclibs();
--- 371,377 ----
  void dump_info_file();
  
  
! void Create_TRD_Geometry_v14a100() {
    // Load the necessary FairRoot libraries 
    gROOT->LoadMacro("$VMCWORKDIR/gconfig/basiclibs.C");
    basiclibs();
