*** Create_TRD_Geometry_v13p_3e.C	2013-11-05 14:38:35.259099932 +0100
--- Create_TRD_Geometry_v13p_3m.C	2013-11-05 14:49:42.487557201 +0100
***************
*** 71,77 ****
  #include <iostream>
  
  // Name of output file with geometry
! const TString geoVersion   = "trd_v13p_3e";
  const TString FileNameSim  = geoVersion + ".geo.root";
  const TString FileNameGeo  = geoVersion + "_geo.root";
  const TString FileNameInfo = geoVersion + ".geo.info";
--- 71,77 ----
  #include <iostream>
  
  // Name of output file with geometry
! const TString geoVersion   = "trd_v13p_3m";
  const TString FileNameSim  = geoVersion + ".geo.root";
  const TString FileNameGeo  = geoVersion + "_geo.root";
  const TString FileNameInfo = geoVersion + ".geo.info";
***************
*** 140,147 ****
  //Double_t LayerPosition[MaxLayers] = { 260. }; // start position - 2013-10-28 - DE - v13x1 - SIS 100 hadron         ( 4 layers, z = 2600 )
  //Double_t LayerPosition[MaxLayers] = { 410. }; // start position - 2013-10-28 - DE - v13x2 - SIS 100 electron       ( 4 layers, z = 4100 )
  //Double_t LayerPosition[MaxLayers] = { 460. }; // start position - 2013-10-28 - DE - v13x3 - SIS 100 muon     3_abs ( 4 layers, z = 4600 )
! Double_t LayerPosition[MaxLayers] = { 410. }; // start position - 2013-10-28 - DE - v13x4 - SIS 300 electron       (10 layers, z = 4100 )
! //Double_t LayerPosition[MaxLayers] = { 550. }; // start position - 2013-10-28 - DE - v13x5 - SIS 300 muon     6_abs (10 layers, z = 5500 )
  //
  // obsolete variants
  //Double_t LayerPosition[MaxLayers] = { 410. }; // start position - 2013-06-25 - DE - v13i trd100_rich             ( 2 layers, z = 4100 )
--- 140,147 ----
  //Double_t LayerPosition[MaxLayers] = { 260. }; // start position - 2013-10-28 - DE - v13x1 - SIS 100 hadron         ( 4 layers, z = 2600 )
  //Double_t LayerPosition[MaxLayers] = { 410. }; // start position - 2013-10-28 - DE - v13x2 - SIS 100 electron       ( 4 layers, z = 4100 )
  //Double_t LayerPosition[MaxLayers] = { 460. }; // start position - 2013-10-28 - DE - v13x3 - SIS 100 muon     3_abs ( 4 layers, z = 4600 )
! //Double_t LayerPosition[MaxLayers] = { 410. }; // start position - 2013-10-28 - DE - v13x4 - SIS 300 electron       (10 layers, z = 4100 )
! Double_t LayerPosition[MaxLayers] = { 550. }; // start position - 2013-10-28 - DE - v13x5 - SIS 300 muon     6_abs (10 layers, z = 5500 )
  //
  // obsolete variants
  //Double_t LayerPosition[MaxLayers] = { 410. }; // start position - 2013-06-25 - DE - v13i trd100_rich             ( 2 layers, z = 4100 )
***************
*** 368,374 ****
  void dump_info_file();
  
  
! void Create_TRD_Geometry_v13p_3e() {
    // Load the necessary FairRoot libraries 
    gROOT->LoadMacro("$VMCWORKDIR/gconfig/basiclibs.C");
    basiclibs();
--- 368,374 ----
  void dump_info_file();
  
  
! void Create_TRD_Geometry_v13p_3m() {
    // Load the necessary FairRoot libraries 
    gROOT->LoadMacro("$VMCWORKDIR/gconfig/basiclibs.C");
    basiclibs();
